### PR TITLE
Fix parser expect messages

### PR DESCRIPTION
**Pull request by hackaugusto**
## Describe your changes

I didn't read too much the code, just noticed some of the expect strings are referring to the wrong expected value. I didn't try to remove the expects.

## Checklist before requesting a review
- Repo forked and branch created from `next` according to naming convention.
- Commit messages and codestyle follow [conventions](./CONTRIBUTING.md).
- Relevant issues are linked in the PR description.
- Tests added for new functionality.
- Documentation/comments updated according to changes.

### CODE REVIEW

File: assembly/src/parsers/context.rs [hunk 1]
```diff
@@ -251,13 +252,13 @@ impl ParserContext {
         // consume the 'end' token
         match tokens.read() {
             None => {
-                let token = tokens.read_at(proc_start).expect("no proc token");
+                let token = tokens.read_at(proc_start).expect("no end token");
```
Comment (bobbinth) [inline assembly/src/parsers/context.rs:255]: I think this and other messages here are actually correct. This line tries to read a token at `proc_start` which should have been a `proc` token.

Comment (hackaugusto) [reply assembly/src/parsers/context.rs:255]: oh, thanks! I didn't notice this was reading at `proc_start`, let me check the messages again.

Comment (hackaugusto) [reply assembly/src/parsers/context.rs]: fixed, it was only the `while` that was using the wrong token name, thanks for getting this!

**Review by bobbinth — APPROVED** [review]
Looks good! Thank you!

**Pull request merged**